### PR TITLE
Don't consider deleted add-ons when determining whether an user is a developer or not

### PR DESCRIPTION
**Pull request by diox**
Fix mozilla/addons/issues/316 again

### CODE REVIEW

**Comment by eviljeff** [comment]
is `user.delete()` a hard delete?  If so what about when we access a review page for a deleted addon if the user has been deleted?

**Comment by diox** [comment]
It is, but we should call it. Instead regular user deletion does `user.anonymize()`, which keeps the relations intact but as the name implies, anonymizes the data. However, both the view and form check if the user is a developer before doing that, which is why we need that fix: otherwise someone who developed an add-on, even a (soft-)deleted one, can never "delete" their account.

File: src/olympia/users/models.py [hunk 1]
```diff
@@ -241,9 +241,10 @@ def picture_url(self):
             ])
             return user_media_url('userpics') + path
 
-    @amo.cached_property
+    @amo.cached_property(writable=True)
```
Comment (eviljeff) [inline src/olympia/users/models.py]: where do we write to the property?  Is this a necessary change?

Comment (diox) [inline src/olympia/users/models.py]: In the tests, it's needed to delete it... that parameter is a particularity of our own implementation (I should really revisit #3396 at some point), django's doesn't have it and happily lets you write/delete cached properties by default.

**Pull request merged**